base/bitarray.jl

```diff
@@ -1020,6 +1064,10 @@ function deleteat!(B::BitVector, inds)
 end
 
 function splice!(B::BitVector, i::Integer)
+    # TODO: after deprecation remove the four lines below
```

Comment:
I have also added the checks for `Bool` to `splice!` as they are done in non-`BitVector` case:
```
julia> splice!([1,2,3], true)
ERROR: ArgumentError: invalid index: true of type Bool

julia> splice!([1,2,3], true, [4,5])
ERROR: ArgumentError: invalid index: true of type Bool
```